wrangler-core/src/main/java/co/cask/directives/parser/ParseTimestamp.java

```diff
@@ -0,0 +1,137 @@
+import co.cask.wrangler.api.DirectiveParseException;
+import co.cask.wrangler.api.ErrorRowException;
+import co.cask.wrangler.api.ExecutorContext;
+import co.cask.wrangler.api.Optional;
+import co.cask.wrangler.api.Row;
+import co.cask.wrangler.api.annotations.Categories;
+import co.cask.wrangler.api.parser.ColumnName;
+import co.cask.wrangler.api.parser.Text;
+import co.cask.wrangler.api.parser.TokenType;
+import co.cask.wrangler.api.parser.UsageDefinition;
+import org.apache.commons.lang3.EnumUtils;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A Executor to parse timestamp as date.
+ */
+@Plugin(type = Directive.Type)
+@Name("parse-timestamp")
+@Categories(categories = {"parser", "date"})
+@Description("Parses column values representing unix timestamp as date.")
+public class ParseTimestamp implements Directive {
+  public static final String NAME = "parse-timestamp";
+  private static final Set<TimeUnit> SUPPORTED_TIME_UNITS = EnumSet.of(TimeUnit.SECONDS, TimeUnit.MILLISECONDS,
+                                                                       TimeUnit.MICROSECONDS);
+  private String column;
+  private TimeUnit timeUnit;
+
+  @Override
+  public UsageDefinition define() {
+    UsageDefinition.Builder builder = UsageDefinition.builder(NAME);
+    builder.define("column", TokenType.COLUMN_NAME);
+    builder.define("timeunit", TokenType.TEXT, Optional.TRUE);
+    return builder.build();
+  }
+
+  @Override
+  public void initialize(Arguments args) throws DirectiveParseException {
+    this.column = ((ColumnName) args.value("column")).value();
+    this.timeUnit = TimeUnit.MILLISECONDS;
+
+    if (args.contains("timeunit")) {
+      String unitValue = ((Text) args.value("timeunit")).value();
+      this.timeUnit = getTimeUnit(unitValue.toUpperCase());
+    }
+  }
+
+  @Override
+  public void destroy() {
+    // no-op
+  }
+
+  @Override
+  public List<Row> execute(List<Row> rows, ExecutorContext context)
+    throws DirectiveExecutionException, ErrorRowException {
+    for (Row row : rows) {
+      int idx = row.find(column);
+      if (idx != -1) {
+        Object object = row.getValue(idx);
+        // If the data in the cell is null or is already of date format, then
+        // continue to next row.
+        if (object == null || object instanceof ZonedDateTime) {
+          continue;
+        }
+
+        long longValue = getLongValue(object);
+        ZonedDateTime zonedDateTime = getZonedDateTime(longValue, timeUnit, ZoneId.ofOffset("UTC", ZoneOffset.UTC));
+        row.setValue(idx, zonedDateTime);
+      }
+    }
+    return rows;
+  }
+
+  private static TimeUnit getTimeUnit(String unitValue) throws DirectiveParseException {
+    if (EnumUtils.isValidEnum(TimeUnit.class, unitValue) &&
```

Comment:
simpler do to something like:

```
TimeUnit unit;
try {
  unit = TimeUnit.valueOf(unitValue.toUpperCase());
} catch (IllegalArgumentException e) {
  throw new DirectiveParseException(...);
}

if (SUPPORTED_TIME_UNITS.contains(unit)) {
  throw new DirectiveParseException(...);
}
```

Reply:
nit: I would uppercase unitValue in this method instead of before you pass it to this method. That way, if there is an error, the user sees the exact value that they specified and not the upper-cased version.